app/javascript/packages/analytics/click-observer-element.ts

```diff
@@ -12,6 +12,19 @@ class ClickObserverElement extends HTMLElement {
     return this.getAttribute('event-name');
   }
 
+  get payload(): object | undefined {
+    const stringPayload = this.getAttribute('data-payload') || undefined;
+    if (stringPayload) {
+      try {
+        return JSON.parse(stringPayload);
+      } catch (e) {
+        return undefined;
+      }
+    } else {
+      return undefined;
+    }
```

Comment:
Nit: I think the `getter-return` ESLint rule is a little aggressive here, but we can still trim some of these branch paths while still accounting for all the different types of values we might expect. We could also use `dataset` to get at data attributes.

```suggestion
    const { payload } = this.dataset;
    if (payload) {
      try {
        return JSON.parse(payload);
      } catch {}
    }

    return undefined;
```